lisa/tools/sysbench.py

```diff
@@ -0,0 +1,481 @@
+        # CPU speed:
+        #   events per second:  2770.19
+        cpu_speed_pattern = re.compile(
+            r"events per second:\s+(?P<event_per_sec>[\d.]+)"
+        )
+        match = cpu_speed_pattern.search(data)
+        if match:
+            result["cpu_speed"] = match.group("event_per_sec")
+
+        return result
+
+    def __process_fileio_perf_result(
+        self,
+        data: str,
+        ops: str,
+    ) -> Dict[Any, Any]:
+        result: Dict[Any, Any] = self.__process_perf_result(data)
+
+        # Sample Output
+        # ================
+        # File operations:
+        #     reads/s:                      2717.15
+        #     writes/s:                     1811.43
+        #     fsyncs/s:                     45.39
+
+        # Throughput:
+        #     read, MiB/s:                  42.46
+        #     written, MiB/s:               28.30
+
+        if ops == "write" or ops == "all":
+            reg_ex_io_per_sec = re.compile(r"writes/s:\s+(?P<write_per_sec>[\d.]+)")
+            io_per_sec = reg_ex_io_per_sec.search(data)
+            if io_per_sec:
+                result["write_io_per_sec"] = io_per_sec.group("write_per_sec")
+            else:
+                result["write_io_per_sec"] = 0
+        if ops == "read" or ops == "all":
+            reg_ex_io_per_sec = re.compile(r"reads/s:\s+(?P<read_per_sec>[\d.]+)")
+            io_per_sec = reg_ex_io_per_sec.search(data)
+            if io_per_sec:
+                result["read_io_per_sec"] = io_per_sec.group("read_per_sec")
+            else:
+                result["read_io_per_sec"] = 0
+
+        reg_ex_fsyncs_per_sec = re.compile(r"fsyncs/s:\s+(?P<fsyncs_per_sec>[\d.]+)")
+        fsyncs_per_sec = reg_ex_fsyncs_per_sec.search(data)
+        if fsyncs_per_sec:
+            result["fsyncs_per_sec"] = fsyncs_per_sec.group("fsyncs_per_sec")
+        else:
+            result["fsyncs_per_sec"] = 0
+
+        if ops == "write" or ops == "all":
+            reg_ex_mib_per_sec = re.compile(
+                r"written, MiB/s:\s+(?P<wr_mib_per_sec>[\d.]+)"
+            )
+            mib_per_sec = reg_ex_mib_per_sec.search(data)
+            if mib_per_sec:
+                result["write_mib_per_sec"] = mib_per_sec.group("wr_mib_per_sec")
+            else:
+                result["write_mib_per_sec"] = 0
+        if ops == "read" or ops == "all":
+            reg_ex_mib_per_sec = re.compile(
+                r"read, MiB/s:\s+(?P<rd_mib_per_sec>[\d.]+)"
+            )
+            mib_per_sec = reg_ex_mib_per_sec.search(data)
+            if mib_per_sec:
+                result["read_mib_per_sec"] = mib_per_sec.group("rd_mib_per_sec")
+            else:
+                result["read_mib_per_sec"] = 0
+
+        return result
+
+    def __send_subtest_msg(
+        self,
+        message_type: Any,
+        test_result: Any,
+        test_name: str,
+        other_fields: Dict[str, Any],
+    ) -> None:
+        subtest_msg = create_perf_message(
```

Comment:
The `tool` of perf messages should be set to "sysbench".

Reply:
done